content/learn/book/your-first-game/_index.md

```diff
@@ -0,0 +1,685 @@
+    let delta_seconds = f32::min(0.2, time.delta_seconds());
+    for (ball, mut transform) in ball_query.iter_mut() {
+        transform.translation += ball.velocity * delta_seconds;
+    }
+}
+```
+This function is turned into a system in our setup function and is called every frame.  
+The first line makes sure that the delta time can never be more than 0.2 seconds.  
+This can happen at the beginning of a game, after some potentially slow loading resources are read from disk.
+Without this line, the ball could potentially "teleport" outside of our game.  
+
+Next we iterate over all instances of type Ball in our game using the familiar Query instance
+and we change the translation (position) using the ball's velocity combined with the delta_seconds.  
+Again we are using the delta_seconds in order to make the game independent of its framerate.
+
+
+```rs
+fn ball_collision_system(
+    commands: &mut Commands,
+    //mut scoreboard: ResMut<Scoreboard>,
+    mut ball_query: Query<(&mut Ball, &Transform, &Sprite)>,
+    collider_query: Query<(Entity, &Collider, &Transform, &Sprite)>,
+) {
+```
+The second function, `ball_collision_system` takes four arguments:
+ * A reference to the Commands instance. We can use it not only to add but also to remove entities. This will become very handy when we add the bricks in the next step.
+ * A commented out instance of our scoreboard. We will need this in the last step of this tutorial.
+ * An instance of a Query that references our Ball and its [Transform](https://docs.rs/bevy/latest/bevy/prelude/struct.Transform.html) and `Sprite`.
+ * An instance of a Query that references all collidable Entities and its [Transform](https://docs.rs/bevy/latest/bevy/prelude/struct.Transform.html), `Collider` and `Sprite`.
+ 
+I won't copy the complete code of the function body here again, since the code is easy enough to understand.
+Basically the code iterates over all balls (of course the game only has one, but this is the way to get it) and then it iterates over all collidable objects.
+The Query object is magic and only returns entities that contain ALL of the elements in the type list.  
+Then the code uses the handy `collide` function we imported at the top of main.rs to check for any collisions.
+If we have a collision, we check if it was with a scorable entity (a brick).  
+If that is the case, we despawn (remove) the entity from the game and increment the score by one.  
+(Since we don't have a scoreboard yet, the line to increment the score is commented out for now.)  
+Then we use some basic math to implement the reflection of the ball and finally we break the loop if we hit a wall.
+
+## Next let's add the Bricks.
+
+Okay, we are almost done! Only two easy steps remain. The first step that remains is adding the bricks. The last step will be to add the scoreboard. Let's go!
+
+Just before our setup function ends, add these lines:
+
+```rs
```

Comment:
TBH, I think we should change this example to have three separate setup functions: one for the bricks, one for the cameras, and one for the scoreboard. IMO monolithic setup functions make the code hard to reason about about and maintain and are not a pattern we should encourage.

Reply:
That decision is above my pay grade :-D